eppo_core/src/event_ingestion/event_dispatcher.rs

```diff
@@ -24,9 +24,31 @@ pub(super) struct EventDispatcherConfig {
     pub max_queue_size: usize,
 }
 
+impl EventDispatcherConfig {
+    pub fn default(sdk_key: String, ingestion_url: String) -> Self {
+        EventDispatcherConfig {
+            sdk_key,
+            ingestion_url,
+            delivery_interval: DEFAULT_DELIVERY_INTERVAL,
+            retry_interval: DEFAULT_RETRY_INTERVAL,
+            max_retry_delay: DEFAULT_MAX_RETRY_DELAY,
+            max_retries: DEFAULT_MAX_RETRIES,
+            batch_size: DEFAULT_BATCH_SIZE,
+            max_queue_size: DEFAULT_MAX_QUEUE_SIZE,
+        }
+    }
+}
+
+const DEFAULT_BATCH_SIZE: usize = 1_000;
+const DEFAULT_DELIVERY_INTERVAL: Duration = Duration::from_millis(10_000);
+const DEFAULT_RETRY_INTERVAL: Duration = Duration::from_millis(5_000);
+const DEFAULT_MAX_RETRY_DELAY: Duration = Duration::from_millis(30_000);
+const DEFAULT_MAX_RETRIES: u32 = 3;
+const DEFAULT_MAX_QUEUE_SIZE: usize = 10_000;
+
 /// EventDispatcher is responsible for batching events and delivering them to the ingestion service
 /// via [`EventDelivery`].
-pub(super) struct EventDispatcher {
+pub struct EventDispatcher {
```

Comment:
I would like to fix `EventDispatcher` type before we make it public.

It's currently meaningless because it's equal to `EventDispatcherConfig`. I would like its constructor to spawn it (similar to `spawn_configuration_poller`) and its impl to expose a nicer API (`track()` instead of a pair of channels). (These two should make integration into SDKs much simpler.)